src/errors.rs

```diff
@@ -68,20 +71,21 @@ impl Serialize for ApiError {
 }
 
 impl<'r> Responder<'r> for ApiError {
-    fn respond_to(self, _request: &Request) -> response::Result<'r> {
-        let mut response = Response::build();
+    fn respond_to(self, request: &Request) -> response::Result<'r> {
+        // figure out the content type we should serialize the error message to. If
+        // content negotiation itself has failed, we just use our preferred one instead
+        let NegotiatedContentType(content_type) = request.guard().succeeded().unwrap_or_default();
```

Comment:
Yeah, a few different things happen in this single line of code :(

Basically, Rocket allows us to access request guards in two different ways:

* in the request handlers they appear as function arguments: the argument data type tells Rocket which guard to instantiate before calling our function; Rockets macros generate the necessary code at compile time behind the scenes
* given a `&Request`, you can get a request guard instance by type, which is what we are doing here. The types are inferred based on the left hand side of =, which is probably what threw you off here. This could have been written as `request.guard::<NegotiatedContentType>()` instead.

Naturally, a guard can either succeed or fail, depending on the contents of a user request and the logic we put into the implementation of the `from_request()` method in our guard. So `succeeded().unwrap_or_default()` here means "give me the instantiated guard if from_request() has succeeded, or the default value of `NegotiatedContentType` if it has failed".

The default value of `NegotiatedContentType` is as if the result of content negotiation was `application/json`.